[Mei an] Duke Increments

meiannn opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

meiannn committed:
Update Duke.java

Set up scanner. Output greeting

meiannn committed:
Update Duke.java

Add echo and exit command

meiannn committed:
Update Duke.java

instantiate arraylist for storing of commands

meiannn committed:
Update Duke.java

add items into list

meiannn committed:
Update Duke.java

adjust output for operation commands

meiannn committed:
bug fixes

fixed some of the syntax errors that caused the file to be unable to compile

meiannn committed:
bug fixes

fixed the code to make it work properly

meiannn committed:
create task class

add task class into folder

meiannn committed:
adjust tasks.java

changed data type of status

meiannn committed:
Update Tasks.java

change status symbols

meiannn committed:
implement done command

added necessary changes to output tasks

meiannn committed:
Update Duke.java

minor changes string manipulation

meiannn committed:
Update Tasks.java

add in tde attributes for tasks

meiannn committed:
Update Duke.java

add to new constructor for tasks

meiannn committed:
Update Tasks.java

adjust task description based on type of task

meiannn committed:
Update Duke.java

add stringbuilder to differentite between different types of tasks

meiannn committed:
Update Tasks.java

change the necessary parts of tasks to fit duke.java

meiannn committed:
bugfix

bugfixes

meiannn committed:
Tasks class fixes

add formatTime method to output the task timing correctly

meiannn committed:
Change Duke.java to differentiate the different tasks

use lastIndexOf instead of stringbuilder

meiannn committed:
compiler generated files

meiannn committed:
added error checking

added error checks for empty todos/deadline/events + nonsensical commands

meiannn committed:
Update Duke.class

meiannn committed:
added in delete function

added in code for the delete command

meiannn committed:
Create EmptyInputException.java

created exception handling for incorrect inputs

ckb055 commented:
Hello, good job on the main class. Perhaps you could extract classes to handle the code in a clearer manner. Such as the Ui class to take in user-input and output, Storage class to store your list of tasks.. etc. Press on :)

meiannn committed:
Delete EmptyInputException.java

Not needed

meiannn committed:
Create Parser.java

Parser file for parsing statements

meiannn committed:
Update Parser.java

Add parse method into parser

meiannn committed:
Update Tasks.java

edited the attributes in Tasks and also the parameters used for constructor

meiannn committed:
Edited Duke.java to make it OOP

Added in the methods and objects from the other java files

meiannn committed:
String formatting for toString method in tasks

edited the string format to fit requirements

meiannn committed:
Create SaveToFile.java

for saving list to txt file

meiannn committed:
Update SaveToFile.java

update code for saving list to TaskList.txt

meiannn committed:
Create TaskList.txt

meiannn committed:
Update Parser.java

added necessary parts for parser to parse commands such as "delete [int]" and "done [int]"

meiannn committed:
Update Duke.java

added in commands done and delete

meiannn committed:
Update Duke.java

Changed duke.java to the most recent one and also added saveToFile object as well as write the task list to the file after every update

meiannn committed:
Update Tasks.java

meiannn committed:
Merge branch 'branch-Level-7'

meiannn committed:
Create TaskList.java

extract task list from duke.java to make it more OOP

meiannn committed:
Make changes to support extraction of tasklist

changes made to both Duke.java and TaskList.java

meiannn committed:
Update .gitignore

ignore changes to txt file

meiannn committed:
Update .gitignore

ignore changes to class files that arise from compiling

meiannn committed:
Update Duke.java

minor changes

meiannn committed:
Update .gitignore

meiannn committed:
Update .gitignore

meiannn committed:
Update Tasks.java

added method to format the timing of the task properly to be printed out

calvincxz commented:
Hi, very interesting approach for this project! I think the logic of the code is generally correct, but your Parser class does not seem to help as much as it still returns a list of strings. Maybe you can transfer the code in your Duke class to the Parser class such that the parse method simply prints the output. This way, you can improve the readability of the Duke class. If you want more OOP, the parse method can return a command object which then prints the output :)

meiannn committed:
Update .gitignore

meiannn committed:
Create Todo.java

todo class extends tasks

meiannn committed:
Create Deadline.java

deadline extends tasks

meiannn committed:
OOP on tasks and different types of tasks

adjust code to let todo extend tasks

meiannn committed:
Update Deadline.java

add constructor and tostring for deadline class

meiannn committed:
Create Event.java

event.java extends tasks.java

meiannn committed:
Create Ui.java

for handling user input and outputs

meiannn committed:
Update Ui.java

Added code to Ui.java to deal with input and output

meiannn committed:
Create Command.java

to handle logic

meiannn committed:
Update Command.java

handles logic of duke

meiannn committed:
Update Tasks.java

edit to allow other extension classes to inherit from tasks.java

meiannn committed:
Update Todo.java

change the constructor to inherit the other common features from tasks.java

meiannn committed:
Update Deadline.java

change the constructor to inherit from tasks.java. Also make the tag a private final

meiannn committed:
Update Event.java

change constructor to inherit from tasks.java and also make tag private final

meiannn committed:
Update Parser.java

edited parser to remove all the logic parts and shift into command

meiannn committed:
Update Duke.java

move to ui

meiannn committed:
Update Duke.java

taken from suggested approach from cs2103 website

meiannn committed:
shift filepath from saveToFile.java to duke.java

meiannn committed:
minor changes

meiannn committed:
Create DukeException.java

for handling exceptions in duke

meiannn committed:
JUnit test files

for JUnit testing

meiannn committed:
Create TodoTest.java

meiannn committed:
Create DeadlineTest.java

meiannn committed:
Create EventTest.java

meiannn committed:
Create SaveToFileTest.java

meiannn committed:
Create ParserTest.java

meiannn committed:
Update Command.java

Changed details to throw duke exception when the input is wrong

meiannn committed:
Update Deadline.java

new constructor to accomodate for loading the list at the start of every program run

meiannn committed:
Update DukeException.java

create DukeException, which is to be thrown in this program whenever there might be user induced exceptions

meiannn committed:
Update Event.java

Overloaded the constructor

meiannn committed:
Update Parser.java

to catch scanner scanning empty string exception

meiannn committed:
Update SaveToFile.java

add load method to load the contents from file whenever the program is ran

meiannn committed:
Update TaskList.java

overload constructor for TaskList.java

meiannn committed:
Update .gitignore

ignore changes to tasklist txt

meiannn committed:
Update Tasks.java

overload constructor for Tasks.java

meiannn committed:
Update Todo.java

Overload constructor for todo.java

meiannn committed:
Update Ui.java

Add in the respective show methods

meiannn committed:
add class files to gitignore

meiannn committed:
Merge branch 'gradle'

meiannn committed:
Update build.gradle

add checkstyle into gradle

meiannn committed:
coding style

change code to conform to coding standards

meiannn committed:
Level-10: GUI

Create GUI for duke

meiannn committed:
add javadocs

meiannn committed:
Level 9: Find

meiannn committed:
let duke recognise datetime

meiannn committed:
B-snooze

add snooze command for duke

meiannn committed:
Merge pull request #4 from hellopanda128/branch-extension

Branch extension

meiannn committed:
Update README.md

meiannn committed:
Create docs:Ui.png

meiannn committed:
no message

meiannn committed:
Update build.gradle

meiannn committed:
Set theme jekyll-theme-cayman

meiannn committed:
Set theme jekyll-theme-cayman

meiannn committed:
bugfix for load and storage

meiannn committed:
Merge branch 'master' of https://github.com/hellopanda128/duke

meiannn committed:
add Ui screenshot
